CMakeLists.txt

```diff
@@ -0,0 +1,585 @@
+# # CMake project for fdk-aac
+#
+# ## Basic usage
+#
+# 1. Download and install CMake >= 3.10 (CMake >= 3.15 is recommended)
+# 2. Clone fdk-aac repository
+# 3. In fdk-aac directory create and go to `build` subdirectory:
+#
+#    mkdir build
+#    cd build
+#
+# 4. Run cmake to configure project with desired build type (Release|Debug):
+#
+#    cmake .. -DCMAKE_BUILD_TYPE=Release
+#
+#    Note CMake configuration without CMAKE_BUILD_TYPE option will not set build
+#    type to some default and it is most likely is not what you want.
+#
+#    If you want to build static (default is shared library), add BUILD_SHARED_LIBS=OFF option:
+#
+#    #    cmake .. -DBUILD_SHARED_LIBS=OFF
+#
+#    You can combine options as well:
+#
+#    cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF
+#
+# 5. Run cmake to build project:
+#
+#    cmake --build . -c Release
+
+cmake_minimum_required(VERSION 3.10)
+
+# Policies
+
+## Enables CMAKE_MSVC_RUNTIME_LIBRARY option support for CMake >= 3.15
+## if you want to use a MSVC multi-threaded statically-linked runtime library
+## If you enable it, CMake will build fdk-acc.dll without external dependencies.
+##
+## Example usage:
+##
+## cmake .. -DCMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$<CONFIG:Debug>:Debug>
+if(POLICY CMP0091)
+  cmake_policy(SET CMP0091 NEW)
+endif()
+
+project(fdk-aac VERSION 2.0.1)
+
+# Includes
+
+include(CheckLibraryExists)
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+if(CMAKE_VERSION VERSION_LESS 3.11)
+  set(CPACK_PACKAGE_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
+  set(CPACK_PACKAGE_VERSION_MINOR ${PROJECT_VERSION_MINOR})
+  set(CPACK_PACKAGE_VERSION_PATCH ${PROJECT_VERSION_PATCH})
+  include(CPack)
+endif()
+
+# Options
+
+option(BUILD_SHARED_LIBS "Build shared library" ON)
+option(BUILD_PROGRAMS "Build aac-enc utility" OFF)
+option(FDK_AAC_INSTALL_CMAKE_CONFIG_MODULE "Install CMake package configuration file" ON)
+option(FDK_AAC_INSTALL_PKGCONFIG_MODULE "Install pkg-config .pc file" ON)
+
+# Checks
+
+## Check if math functions are in separate library (Most of Linux distros, maybe some other OSes)
+find_library(LIBM m)
+if (LIBM)
+  check_library_exists(m sin "" HAVE_LIBM)
```

Comment:
This is still the other way around from what I thought - I meant to first check if one can link `sin` without adding anything, and if that doesn't work, see if the `m` library exists and if that fixes linking `sin`.

If it ends up too clumsy it's not necessarily worth it though.

Reply:
Hmm, i was tested it before in other projects and it works. this one is checked on MSYS2 MinGW64 where `check_library_exists(m sin "" HAVE_LIBM)` gives false positive. I guess it is better to leave it as it is.

Reply:
The fact that it seems to work on msys2 mingw is because cmake doesn't look in /mingw64/x86_64-w64-mingw32/lib by default, for some reason, while the linker itself checks that directory (which is why it is found if you just do `check_library_exists` straight away).

Something like this should do more of what I wanted:
```
include(CheckFunctionExists)
[...]
check_function_exists(sin HAVE_DEFAULT_MATH)
if (NOT HAVE_DEFAULT_MATH)
  check_library_exists(m sin "" HAVE_LIBM)
endif()
```
That has the right semantics IMO, and isn't any more complicated.

Reply:
> The fact that it seems to work on msys2 mingw is because cmake doesn't look in /mingw64/x86_64-w64-mingw32/lib by default, for some reason, while the linker itself checks that directory (which is why it is found if you just do `check_library_exists` straight away).

(Also if one would do e.g. run cmake with `-DCMAKE_LIBRARY_PATH=/mingw64/x86_64-w64-mingw32/lib`, then it would find and use libm again.)

Reply:
That sounds reasonable.